src/ansys/fluent/core/launcher/launcher_utils.py

```diff
@@ -615,12 +614,11 @@ def launch_remote_fluent(
         launcher_args=launcher_args,
     )
 
-    remote_file_handler = RemoteFileHandler(
-        transfer_service=PimFileTransferService(
+    return session_cls(
+        fluent_connection=fluent_connection,
+        remote_file_handler=remote_file_handler(
             pim_instance=fluent_connection._remote_instance
         )
-    )
-
-    return session_cls(
-        fluent_connection=fluent_connection, remote_file_handler=remote_file_handler
+        if pypim.is_configured()
+        else remote_file_handler(),
```

Comment:
@hpohekar at this point, the `file_transfer_service` is already selected and no further logic is needed. Also, why `remote_file_handler()` and not just `remote_file_handler` (should be named `file_transfer_service`).

Reply:
@seanpearsonuk  It's not clear to me - "at this point, the file_transfer_service is already selected and no further logic is needed." 

"why remote_file_handler() and not just remote_file_handler" - If file_transfer_service is PimFileTransferService it needs pim_instance specifically when  we use launch_remote_fluent to launch Fluent in Ansys Lab.

```
file_transfer_service  = PimFileTransferService(pim_instance=fluent_connection._remote_instance) or 
file_transfer_service  = MyFileService()
```

Therefore we have written it as follows -

```
remote_file_handler = (
        file_transfer_service(pim_instance=fluent_connection._remote_instance)
        if pypim.is_configured()
        else file_transfer_service()
    )

return session_cls( fluent_connection=fluent_connection, remote_file_handler=remote_file_handler)
```

Reply:
@seanpearsonuk I understood this now. Removed the logic and removed the call to `file_transfer_service`. Thank you.

Reply:
We have kept default value None and we are instantiating it inside the body of the function if it's None.